develop/blockchains/deployment/generate-chain-specs.md

```diff
@@ -1,4 +1,136 @@
+- `properties` - custom properties provided as a key-value JSON object
+- `codeSubstitutes` - an optional mapping of block numbers to Wasm code
+- `genesis` - the genesis configuration for the chain
+
+For example, the following JSON shows a basic chain specification file:
+
+```json
+--8<-- "code/develop/blockchains/deployment/generate-chain-specs/basic-chain-specs.json"
+``` 
+
+
+## Creating a Custom Chain Specification
+
+To create a custom chain specification, you can use the [`chain-spec-builder`](https://paritytech.github.io/polkadot-sdk/master/staging_chain_spec_builder/index.html){target=\_blank} tool. This is a CLI tool that is used to generate chain specifications from the runtime of a node. To install the tool, run the following command:
+
+```bash
+cargo install staging-chain-spec-builder
+```
+
+To verify the installation, run the following:
+
+```bash
+chain-spec-builder --help
+```
+
+### Plain Chain Specifications
+
+To create a plain chain specification, you can use the following utility within your project:
+
+```bash
+chain-spec-builder create -r <RUNTIME_WASM_PATH> <INSERT_COMMAND> 
+```
+
+!note
+    Before running the command, ensure that the runtime has been compiled and is available at the specified path.
+
+Ensure to replace `<RUNTIME_WASM_PATH>` with the path to the runtime Wasm file and `<INSERT_COMMAND>` with the command to insert the runtime into the chain specification. The available commands are:
+
+- `patch` - patches the runtime's default genesis config with the provided patch
+- `full` - build the genesis config for runtime using the JSON file. No defaults will be used
+- `default` - gets the default genesis config for the runtime and uses it in `ChainSpec`. Please note that the default genesis config may not be valid. For some runtimes, initial values should be added there (e.g., session keys, babe epoch)
+- `named-preset` - uses named preset provided by the runtime to build the chains spec
+
+### Raw Chain Specifications
+
+With runtime upgrades, the blockchain's runtime can be upgraded with newer business logic. Chain specifications contain information structured in a way that the node's runtime can understand. For example, consider this excerpt of a common entry for a chain specification:
+
+```json
+"sudo": {
+  "key": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY"
+}
+```
+
+This key and its associated value are human-readable text in the JSON file. However, this information can't be stored in this format in Substrate’s underlying storage structures. Before you can use the chain specification to initialize the genesis storage for a node, the human-readable keys must be transformed into actual storage keys that allow the values to be stored in the storage trie. This transformation is straightforward but requires the chain specification to be encoded in a format that node runtime can read.
```

Comment:
Actually the plain, human readable format, can be used for initialization of the storage, but this involves converting the plain values into the trie storage items by the runtime.

We use raw format if we want to avoid this conversion - e.g. for long-live chains like testnets or production, to avoid potential changes in this conversion process which may happen when runtime is upgraded.

I'd rephrase this paragraph to make it more clear. IMO, currently it is slightly misleading.

Reply:
Great! I rephrased it here - [94e3a4c](https://github.com/polkadot-developers/polkadot-docs/pull/27/commits/94e3a4cccfbd8b3c77e2809a05fe9b97d0941653)

Would you mind looking at it to see if it makes more sense now? @michalkucharczyk

Reply:
Perfect. Thank you.